integration/test/performancetest/transmitter.go

```diff
@@ -205,44 +206,62 @@ func (transmitter *TransmitterAPI) Parser(data []byte) (map[string]interface{},
 	}
 	packet := make(map[string]interface{})
 	packet[PARTITION_KEY] = time.Now().Year()
-	packet[HASH] =  os.Getenv(SHA_ENV) //fmt.Sprintf("%d", time.Now().UnixNano())
+	packet[HASH] = os.Getenv(SHA_ENV) //fmt.Sprintf("%d", time.Now().UnixNano())
 	packet[COMMIT_DATE],_ = strconv.Atoi(os.Getenv(SHA_DATE_ENV))
-	
+
 	for _, rawMetricData := range dataHolder {
-		numDataPoints := float64(len(rawMetricData.Timestamps))
-		var avg float64
-		max := 0.0
-		min := 10000.0
-		if numDataPoints <= 0{
-			avg = 0.0
-			max = 0.0
-			min = 0.0
-			log.Fatalf("Error there is no data points")
-		}else{
-			// @TODO:ADD GetMetricStatistics after merging with data collection code
-			sum :=0.0
-			for _,val := range rawMetricData.Values {
-				sum += val
-				if max < val{
-					max = val
-				}
-				if min > val {
-					min = val
-				}
-			}
-			
-			avg =  sum /numDataPoints
-		}
-		//----------------
-		metric := Metric{
-			Average: avg,
-			Max:     max,
-			Min:     min,
-			P99:     0.0,
-			Std: 	rand.Float64(),
-			Period:  int(METRIC_PERIOD / (numDataPoints)),
-			Data:    rawMetricData.Values}
+
+		metric := CalcStats(rawMetricData.Values)
+
 		packet[rawMetricData.Label] = metric
 	}
 	return packet, nil
-}
\ No newline at end of file
+}
+
+//CalcStats takes in an array of data and returns the average, min, max, p99, and stdev of the data in a Metric struct
+func CalcStats(data []float64) Metric {
+	length := len(data)
+	if length == 0 {
+		return Metric{}
+	}
+
+	//make a copy so we aren't modifying original
+	dataCopy := make([]float64, length)
```

Comment:
We are modifying the data locally so why we don't modify the original ? Is the original data being used by somewhere else?